### ISSUE TITLE
OPDS catalog link not working for OC 9.0

### ISSUE THREAD

**Issue by filipkis** [Yetangitu/owncloud-apps#32]
When installing this plugin on OC 9.0 the personal catalog link does not work. It requests authentication, but after correct credentials are entered it fails. This is the information from the log:

    Call to undefined method OC\User\Session::logClientIn() at /var/www/owncloud/apps_other/files_opds/lib/util.php#84

I quickly checked the source of OC and indeed the logClientIn method has been changed between 9.0 and 9.1 version. So I assume this is the cause.

**Comment by Yetangitu** [comment]
Hm, strange. It works here with OC 9.1.3.1. OPDS Catalog uses the method logClientIn with three parameters on Owncloud, four on Nextcloud. Did you just upgrade OC? Did you restart your PHP instance or webserver?

**Comment by filipkis** [comment]
So, sorry for not being clear. I had OC 9.0.something (don't remember which
exact subversion) and it didn't work there. I then upgraded to OC 9.1.4 and
it worked. And then checking the repository of OC:

https://github.com/owncloud/core/blame/f10d105ea2f2d0822693313ddcd6f9b2eaac6ebe/lib/private/User/Session.php

I see that the logClientIn was added recently. And indeed if you check the
latest OC 9.0.7 code:

https://github.com/owncloud/core/blob/v9.0.7/lib/private/user/session.php

The function is not there. I might not have been looking in the right
place, but the fact it it didn't work on my 9.0 instance and upgrading to
9.1.4 fixed it.

Filip
